### ISSUE TITLE
Understand the training output

### ISSUE THREAD

**Issue by pandasMX** [AlexeyAB/darknet#668]
Hi, I'm training on my own dataset with 90 categories using yolov2. I'm trying to understand the training output. For example,

Region Avg IOU: 0.647465, Class: 0.040123, Obj: 0.442353, No Obj: 0.012484, Avg Recall: 0.666667,  count: 3
Region Avg IOU: 0.770215, Class: 0.006021, Obj: 0.589292, No Obj: 0.009339, Avg Recall: 1.000000,  count: 2
Region Avg IOU: 0.754976, Class: 0.006246, Obj: 0.305856, No Obj: 0.012812, Avg Recall: 1.000000,  count: 6
Region Avg IOU: 0.778233, Class: 0.028962, Obj: 0.441049, No Obj: 0.009935, Avg Recall: 1.000000,  count: 3
Region Avg IOU: 0.660372, Class: 0.024296, Obj: 0.246137, No Obj: 0.011715, Avg Recall: 1.000000,  count: 2
51458: 9.307969, 10.018262 avg, 0.000004 rate, 1.161277 seconds, 3521890 images

It's been trained quite a long time, around 1 day, but I see the loss is still very high. The IOU and recall seem to be good. However, I don't understand the meaning of class, Obj, No Obj. But from my experience, the value of class should be around 1 if the training goes well. It's still very low until now though. Can you take a look and maybe give me some advises? Thanks!

**Comment by AlexeyAB** [comment]
Hi, yes, your avg loss is high for 50 000 iterations, probably due to low learning rate `0.000004 rate`. Try to change `learning_rate=, scales=, steps=` in your cfg-file.
1 day isn't a long time if you don't use Tesla V100 or DGX :)

Output of Yolo v3: https://github.com/AlexeyAB/darknet/issues/636#issuecomment-381400954
* `Obj:` - is average of the objectness `T0` (probabilities that there is an object in this cell)
* `No Obj:` - is average of the probabilities of all objects (both correctly and not correctly classified)

**Comment by pandasMX** [comment]
Oh, the learning rate became 0 is actually caused by the steps, and scales,
```
learning_rate=0.00004
burn_in=1000
max_batches = 80200000
policy=steps
steps=40000,60000
scales=.1,.1
```

**Comment by pandasMX** [comment]
Thanks a lot! One more question, what's the meaning of 'count' ? I set the batch to 64 and subdivisions to 32. My understanding is that each GPU will take one batch at a time and the 64 images will be divided into 32 groups. Therefore it will print 32 lines each time. And the count in theory should be <= 2 for each line because 64/32 = 2. But I see the count can be larger than 2 sometimes. Maybe my understanding is wrong?

**Comment by ralek67** [comment]
Count is the number of labelled objects present in the images processed during this line of training.
Batch = 64 means it takes 64 images to train in your train.txt per iteration. Subdivisions = 32 means it process 2 images per line. 
So if it uses an image where you labelled 40 objects and another with 3 it will say count: 43

Correct me if i'm wrong.

**Comment by pandasMX** [comment]
@ralek67 Thank you! That sounds correct.

**Comment by AlexeyAB** [comment]
@pandasMX 

> And do you know why the class value is so low?

* `Class:` - is average of the probabilities of correctly classified objects

For some reason, detected objects have very low probability.

----

* What params do you use in the Makefile? https://github.com/AlexeyAB/darknet/blob/5d616450a43f4b5fdf130f57cdaab105cf2563e2/Makefile#L1-L7

* What params do you use in your cfg? https://github.com/AlexeyAB/darknet/blob/5d616450a43f4b5fdf130f57cdaab105cf2563e2/cfg/yolov3.cfg#L18-L23

**Comment by pandasMX** [comment]
@AlexeyAB The params in Makefile:
```
 GPU=1
 CUDNN=1 
 CUDNN_HALF=0 
 OPENCV=0 
 AVX=0 
 OPENMP=0 
 LIBSO=0 
```
I changed the learning rate to 0.004 now.
The params in yolov2-voc.cfg
```
 learning_rate=0.004 
 burn_in=1000 
 max_batches = 8000000
 policy=steps 
 steps=4000,6000
 scales=.1,.1
```
I'm not pretty sure.But for the low class values, my guess is that I have too many categories and they are hard for the classifier to learn. My IOU and Obj values now are still low, and I think yolo will learn to locate the object first and only after that it will be able to learn the features to distinguish the different classes.

**Comment by ralek67** [comment]
@pandasMX change steps !!!
for 80 classes it is 400 000 / 450 000 for 500 000 max batches so use equivalently.
you get 0.00004 LR after 6000 iterations that's why

**Comment by pandasMX** [comment]
@ralek67 Thanks! This might be the problem. I'll increase the learning rate.

**Comment by AlexeyAB** [comment]
Use this in your cfg-file
```
 learning_rate=0.001
 burn_in=1000 
 max_batches = 8000000
 policy=steps 
 steps=400000,450000 
 scales=.1,.1
```

----

Also if you have installed OpenCV and you use Tesla V100, then use these params in the Makefile - training will be 2x faster:
```
 GPU=1
 CUDNN=1
 CUDNN_HALF=1
 OPENCV=1
 AVX=0 
 OPENMP=0 
 LIBSO=0 
```

**Comment by pandasMX** [comment]
Thanks @AlexeyAB and @ralek67 , the problem has been solved after I increased the learning rate.

**Comment by rafaelpadilla** [comment]
@AlexeyAB you said:
"Obj: - is average of the objectness T0 (**probabilities that there is an object in this cell**)"

According to the paper, the objectness is calculated by each bounding box, not the cell.
Each bounding box has its own objectness. Each cell has 5 bounding box.

Regards,
Mr. Pad

**Comment by AlexeyAB** [comment]
@rafaelpadilla Yes, thanks, I fixed it.
